### PR TITLE
#5492: Mixpanel - add logging for groups/gating

### PR DESCRIPTION
**Pull request by Miaplacidus**
Log events listed below in Mixpanel. 

- Members Page
  - [x] Page Viewed: feature usage
- Groups Tab on Members Page
  - [x] Page Viewed: feature usage
- Create Group Page
  - [x] Page viewed: admin interest in feature, # of page views
  - [x] Groups created: follow-through on form, usage, # of form completions
- Edit Group Page
  - [x] Page Viewed: # of page views, interest in updates.
  - [x] Successful Edits: # of successful updates, follow-through + completions
- Informative Banner Clickthrough to Groups Tab on Members Page
  - [x] From View Thread Page (when trying to comment on a thread in a gated topic): # of impacted users who want more info

The following task was not completed as there is no banner clickthrough on the new thread page:
  - From Create Thread Page (when trying to post in a gated topic): # of impacted users who don't complete the thread

## Link to Issue
Closes: #5492 

## Description of Changes
- log events listed above in Mixpanel

## Test Plan
- To test locally, make sure that the variables in your `.env` referring to your node environment, the gating api feature flag, the mixpanel token, and mixpanel dev token have correct values. Please see the Mixpanel Common project for the token.

### CODE REVIEW

**Review by mzparacha — APPROVED** [review]

File: packages/commonwealth/test/unit/server_controllers/server_groups_controller.spec.ts [hunk 1]
```diff
@@ -172,7 +172,7 @@ describe('ServerGroupsController', () => {
   describe('#createGroup', async () => {
     const controller = createMockedGroupsController();
     const { user, chain, address } = createMockParams();
-    const result = await controller.createGroup({
+    const [result] = await controller.createGroup({
```
Comment (rbennettcw) [inline packages/commonwealth/test/unit/server_controllers/server_groups_controller.spec.ts:175]: Can we add an assertion that the analytics object contains the correct event data here?

It's one of the benefits of decoupling event creation from event publishing

File: packages/commonwealth/test/unit/server_controllers/server_groups_controller.spec.ts [hunk 2]
```diff
@@ -210,7 +210,7 @@ describe('ServerGroupsController', () => {
   describe('#updateGroup', async () => {
     const controller = createMockedGroupsController();
     const { user, chain, address } = createMockParams();
-    const result = await controller.updateGroup({
+    const [result] = await controller.updateGroup({
```
Comment (rbennettcw) [inline packages/commonwealth/test/unit/server_controllers/server_groups_controller.spec.ts:213]: Also assert here please

**Comment by rbennettcw** [comment]
looks good– just has conflict

**Comment by Miaplacidus** [comment]
@rbennettcw Conflict resolved!

**Review by rbennettcw — APPROVED** [review]

**Pull request merged**